app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -11,6 +11,27 @@ module TwoFactorAuthenticatable
   DIRECT_OTP_VALID_FOR_SECONDS = DIRECT_OTP_VALID_FOR_MINUTES * 60
   REMEMBER_2FA_COOKIE = 'remember_tfa'.freeze
 
+  class AuthMethod
+    BACKUP_CODE = 'backup_code'
+    PERSONAL_KEY = 'personal_key'
+    PIV_CAC = 'piv_cac'
+    REMEMBER_DEVICE = 'remember_device'
+    SMS = 'sms'
+    TOTP = 'totp'
+    VOICE = 'voice'
+    WEBAUTHN = 'webauthn'
+    WEBAUTHN_PLATFORM = 'webauthn_platform'
+    PHISHING_RESISTANT_METHODS = [
+      WEBAUTHN,
+      WEBAUTHN_PLATFORM,
+      PIV_CAC,
+    ].freeze
```

Comment:
style/micro-optim:
1. add an extra line to separate
2. make it a set because we're checking membership, not using order for anything

```suggestion

    PHISHING_RESISTANT_METHODS = [
      WEBAUTHN,
      WEBAUTHN_PLATFORM,
      PIV_CAC,
    ].to_set.freeze
```